### PR TITLE
Upgrade kubernetes-client to 4.3.0

### PR DESCRIPTION
**Pull request by carlossg**

### CODE REVIEW

**Comment by carlossg** [comment]
Blocked by https://github.com/fabric8io/kubernetes-client/pull/1500

File: pom.xml [hunk 1]
```diff
@@ -48,13 +48,14 @@
...
     <pipeline-model-definition.version>1.3.7</pipeline-model-definition.version>
+    <slf4jVersion>1.7.26</slf4jVersion>
   </properties>
 
   <dependencies>
     <dependency>
       <groupId>io.fabric8</groupId>
       <artifactId>kubernetes-client</artifactId>
-      <version>4.1.3</version>
+      <version>4.2.0</version>
```
Comment (jglick) [inline pom.xml]:
```suggestion
      <version>4.2.2</version>
```

File: src/main/java/org/csanchez/jenkins/plugins/kubernetes/pipeline/ContainerLogStepExecution.java [hunk 2]
```diff
@@ -72,9 +72,9 @@ protected String run() throws Exception {
 
             String podName = nodeContext.getPodName();
             ContainerResource<String, LogWatch, InputStream, PipedOutputStream, OutputStream, PipedInputStream,
-                    String, ExecWatch> container = client.pods()
-                    .inNamespace(nodeContext.getNamespace())
-                    .withName(podName)
+                    String, ExecWatch, Boolean, InputStream> container = client.pods() //
```
Comment (jglick) [inline src/main/java/org/csanchez/jenkins/plugins/kubernetes/pipeline/ContainerLogStepExecution.java]:
```suggestion
                    String, ExecWatch, Boolean, InputStream> container = client.pods()
```

File: src/main/java/org/csanchez/jenkins/plugins/kubernetes/pipeline/ContainerLogStepExecution.java [hunk 3]
```diff
@@ -72,9 +72,9 @@ protected String run() throws Exception {
...
+                    .inNamespace(nodeContext.getNamespace()) //
```
Comment (jglick) [inline src/main/java/org/csanchez/jenkins/plugins/kubernetes/pipeline/ContainerLogStepExecution.java]:
```suggestion
                    .inNamespace(nodeContext.getNamespace())
```

File: src/main/java/org/csanchez/jenkins/plugins/kubernetes/pipeline/ContainerLogStepExecution.java [hunk 4]
```diff
@@ -72,9 +72,9 @@ protected String run() throws Exception {
...
+                    .withName(podName) //
```
Comment (jglick) [inline src/main/java/org/csanchez/jenkins/plugins/kubernetes/pipeline/ContainerLogStepExecution.java]:
```suggestion
                    .withName(podName)
```

**Comment by jglick** [comment]
> Blocked by [fabric8io/kubernetes-client#1500](https://github.com/fabric8io/kubernetes-client/pull/1500)

Thought it seems that it still not in 4.2.2?

**Comment by jglick** [comment]
All tests failing with

```
java.lang.NullPointerException
	at org.csanchez.jenkins.plugins.kubernetes.pipeline.ContainerExecDecorator$1.doLaunch(ContainerExecDecorator.java:377)
	at org.csanchez.jenkins.plugins.kubernetes.pipeline.ContainerExecDecorator$1.launch(ContainerExecDecorator.java:252)
	at hudson.Launcher$ProcStarter.start(Launcher.java:454)
	at org.jenkinsci.plugins.durabletask.BourneShellScript.launchWithCookie(BourneShellScript.java:188)
	at …
```

which appears to mean that `ExecWebSockerListener.getInput()` is returning null. Of course the interface method has no Javadoc.

I would not mind switching to the [official client](https://github.com/kubernetes-client/java). Have had better experiences with it.

**Comment by Vlatombe** [comment]
4.3.0 released, need to update this PR

**Pull request merged**